lib/Parse/ParseStmt.cpp

```diff
@@ -1522,6 +1522,10 @@ Parser::parseAvailabilitySpecList(SmallVectorImpl<AvailabilitySpec *> &Specs,
       consumeToken();
       Status.setIsParseError();
     } else if (consumeIf(tok::comma)) {
+      // End of list with a trailing comma.
+      if (Source != AvailabilitySpecSource::Available && Tok.is(tok::r_paren)) {
+        break;
+      }
```

Comment:
Why do we need to check for `Source != AvailabilitySpecSource::Available` here?

Reply:
I did this so `#available` won't support trailing comma. It doesn't really make sense and it leads to worse diagnostic when `*` is missing.

Reply:
Why don’t we need to check for `AvailabilitySpecSource::Unavailable`?

Reply:
Makes more sense indeed.